source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,166 @@
+
+=============
+Serialization
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about serialization with the {+driver-long+}.
+Serialization is the process of mapping a C# object to and from a BSON document.
+
+Serializer Registry
+-------------------
+
+The serializer registry is an implementation of ``IBsonSerializerRegistry``
+that contains all registered serializers. You can access the serializer registry
+with the ``SerializerRegistry`` property of the ``BsonSerializer``
+class as follows:
+
+.. code-block:: csharp
+
+   var intSerializer = BsonSerializer.SerializerRegistry.GetSerializer<int>();
+
+.. important::
+
+   The serializer registry is a global registry. This means that you cannot use
+   multiple registries in a single application.
+
+Serialization Provider
+----------------------
+
+A serialization provider allows you to provide serializers to your
+serialization registry. The serialization provider should be registered as early
+as possible in your application to ensure that the serializers are used by the
+registry. Your serialization provider must implement the
+``IBsonSerializationProvider`` interface. 
+
+Example
+~~~~~~~
+
+The following example implements a serialization provider. The provider has a
+``GetSerializer`` method that returns a serializer based on the type of the
+input. If the method input is an ``int`` it returns a custom ``Int32``
+serializer, otherwise it returns ``null`` so that it can be handled by another
+provider. 
+
+.. code-block:: csharp
+   
+   class MyProvider : IBsonSerializationProvider
+   {
+       public IBsonSerializer GetSerializer(Type type)
+       {
+           if (type == typeof(int))
+           {
+               return new MyInt32Serializer();
+           }        
+   
+           return null;
+       }
+   }
+
+You can register the serialization provider using the
+``RegisterSerializationProvider`` method as shown in the following example:
+
+.. code-block:: csharp
+
+   BsonSerializer.RegisterSerializationProvider(new MyProvider);
+
+Serializers
+-----------
+
+Serializers are classes that handle the translation of C# objects to and
+from BSON documents. Most serializers implement the ``IBsonSerializer``
+interface. The {+driver-short+} has many built-in serializers made to handle
+primitive types, collection types, and custom classes.
+
+For a full list of available serializers, see the
+`Serializers namespace API documentation <{+api-root+}/N_MongoDB_Bson_Serialization_Serializers.htm>`__.
+
+Custom Serializers
+~~~~~~~~~~~~~~~~~~
+
+In some cases, you may need to implement a custom serializer. When creating a
+custom serializer, your class should inherit from ``SerializerBase<T>`` and
+override the ``Deserialize()`` and ``Serialize()`` methods. 
+
+The following code example shows how to implement a custom serializer for
+an ``Int32``:
```

Comment:
S: I assume that anyone reading this page knows C# well enough to understand that "int" is an alias for "Int32", but maybe not a bad idea to just use "int" everywhere.

```suggestion
an ``int``:
```